Grbl_Esp32/src/Serial.cpp

```diff
@@ -85,83 +85,80 @@ void heapCheckTask(void* pvParameters) {
     }
 }
 
-void serial_init() {
+void client_init() {
 #ifdef DEBUG_REPORT_HEAP_SIZE
     // For a 2000-word stack, uxTaskGetStackHighWaterMark reports 288 words available
     xTaskCreatePinnedToCore(heapCheckTask, "heapTask", 2000, NULL, 1, NULL, 1);
 #endif
 
-    Serial.begin(BAUD_RATE);
-    Serial.setRxBufferSize(256);
-    // reset all buffers
-    serial_reset_read_buffer(CLIENT_ALL);
-    grbl_send(CLIENT_SERIAL, "\r\n");  // create some white space after ESP32 boot info
-    serialCheckTaskHandle = 0;
+    Uart0.begin(BAUD_RATE, SERIAL_8N1, 3, 1, false);  // Tx on pin 3, Rx on pin 1 - the standard hardware pins
+    client_reset_read_buffer(CLIENT_ALL);
+    Uart0.write("\r\n");  // create some white space after ESP32 boot info
+    clientCheckTaskHandle = 0;
     // create a task to check for incoming data
     // For a 4096-word stack, uxTaskGetStackHighWaterMark reports 244 words available
     // after WebUI attaches.
-    xTaskCreatePinnedToCore(serialCheckTask,    // task
-                            "serialCheckTask",  // name for task
+    xTaskCreatePinnedToCore(clientCheckTask,    // task
+                            "clientCheckTask",  // name for task
                             4096,               // size of task stack
                             NULL,               // parameters
                             1,                  // priority
-                            &serialCheckTaskHandle,
+                            &clientCheckTaskHandle,
                             SUPPORT_TASK_CORE  // must run the task on same core
```

Comment:
To avoid the vague SMP issues mentioned in the documentation, we should probably ensure that all serial communication to a single uart uses the same core. Currently, the real-time commands use core 1, while the other commands use core 0. Should be an easy fix though, just change the serial task to use the same core. I would guess there is probably plenty of time left on core 0 to do a bit of serial comms.